### PR TITLE
local ip bind to service

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- Features
for security purposes,  local services should bind to local IP but not 0.0.0.0 or ::
- Bug fixes
- Docs
- Tests

when deploying below deployment or ds :kube-ovn-controler/kube-ovn-monitor/kube-ovn-cni/ovn-central

add an env variable
            - name: ENABLE_BIND_LOCAL_IP
              value: "true"

it will work like below
```
root@kube-ovn-worker:/kube-ovn# netstat -tunlp | grep kube-ovn
tcp6       0      0 fc00:f853:ccd:e79:10665 :::*                    LISTEN      1805/./kube-ovn-dae
tcp6       0      0 fc00:f853:ccd:e79:10660 :::*                    LISTEN      1468/./kube-ovn-con
```
this feature will not work when the network is in the dual mode because golang can't support binding v4 and v6 addresses at the same time.

#### Which issue(s) this PR fixes:
Fixes #(issue-number)
in jira ACP-20335

### CODE REVIEW

File: cmd/controller_health_check/controller_health_check.go [hunk 1]
```diff
@@ -23,7 +24,22 @@ func CmdMain() {
 	if err := ovs.CheckAlive(); err != nil {
 		os.Exit(1)
 	}
-	conn, err := net.DialTimeout("tcp", "127.0.0.1:10660", 3*time.Second)
+
+	addr := "127.0.0.1:10660"
```
Comment (oilbeater) [inline cmd/controller_health_check/controller_health_check.go:28]: health_check should only bind to localhost

File: dist/images/install.sh [hunk 2]
```diff
@@ -2023,6 +2023,10 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
```
Comment (oilbeater) [inline dist/images/install.sh:2025]: should be able to set ENABLE_BIND_LOCAL_IP from install.sh and helm

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: cmd/controller_health_check/controller_health_check.go (hunk 1 above)
Comment (changluyi) [reply cmd/controller_health_check/controller_health_check.go:28]: if kube-ovn-controller bind local IP, eg 172.18.0.2:10660,   the controller health check service will get the fault "connection refused " when accessing the 127.0.0.1 10660

Comment (oilbeater) [reply cmd/controller_health_check/controller_health_check.go:28]: ok

File: dist/images/install.sh (hunk 2 above)
Comment (changluyi) [reply dist/images/install.sh:2025]: I set ENABLE_BIND_LOCAL_IP = true, on another pr , and it seems ok when running workflows

Comment (oilbeater) [reply dist/images/install.sh:2025]: This option should be configurable

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: dist/images/install.sh (hunk 2 above)
Comment (changluyi) [reply dist/images/install.sh:2025]: it can be on/off, the pod will restart using a new env value.

Comment (changluyi) [reply dist/images/install.sh:2026]: fixed

**Review by oilbeater — APPROVED** [review]

**Pull request merged**